x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/termenum/action/TermEnumAction.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.core.termenum.action;
+
+import org.elasticsearch.action.ActionType;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+
+import static org.elasticsearch.index.query.AbstractQueryBuilder.parseInnerQueryBuilder;
+
+public class TermEnumAction extends ActionType<TermEnumResponse> {
```

Comment:
Should we use the plural (Terms) instead ? That would be consistent with the name of the API (`_terms`). Same for all the other class names (`TermEnumResponse`, ...).

Reply:
I don't know if "Terms" will be too short. We already have a lot of `o.e.Term*` classes to navigate and users looking for help with the `terms` aggregation. 
A verb would be a good addition e.g. termFinder

Reply:
No have REST endpoint `_terms_enum` and `TermsEnumXxxx` class names